### PR TITLE
Add auditor client capability (Base)

### PR DESCRIPTION
**Pull request by jnmt**
This patch provides auditor client capability. In addition, it includes some refactorings for serialized binary interfaces to put the common logic into one place (base side). See also https://github.com/scalar-labs/scalardl-node-client-sdk/pull/39.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -98,6 +112,10 @@ class ClientServiceBase {
    */
   async registerCertificate() {
     const request = await this._createCertificateRegistrationRequest();
+    return this._registerCertificate(request);
```
Comment (jnmt) [inline index.js:115]: @feeblefakie As we discussed offline, this change is intended to call from Node.js SDK using serialized binary I/F while keeping the main logic commonized here. See also,
https://github.com/scalar-labs/scalardl-node-client-sdk/pull/39/files#diff-329531d245e95e56f45d166c536ccee5fcd533cc3ca870c2fe5b75483e20e3c3R142.

**Review by Torch3333 — APPROVED** [review]
Sorry for the late review. 
LGTM!

**Comment by jnmt** [comment]
Just updated the I/F name since auditor's specification will be changed.  This successfully ran with the latest scalar.proto in my local Node.js/Web SDK environment. PTAL.

**Comment by jnmt** [comment]
Sorry, I found that we should validate the response from the auditor if the response has proof(s). Please wait until the fix pushed.

**Comment by jnmt** [comment]
@feeblefakie @Torch3333 Now ready to review again. PTAL. Thank you.

**Review by Torch3333 — COMMENTED** [review]
Thanks, the code looks good. 
Should unit test be added to cover the newly added logic ?

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Let's add unit tests for the newly added logic after we work on benchmarking.

**Pull request merged**